apps/frontend/src/components/global/ExportHTMLModal.vue

```diff
@@ -119,22 +119,60 @@ export default class ExportHTMLModal extends Vue {
     }
 
     const files = [];
+    const filteredOn = this.filter.status;
+
     for (const fileId of this.filter.fromFile) {
       const file = InspecDataModule.allEvaluationFiles.find(
         (f) => f.uniqueId === fileId
       );
+
       if (file) {
         const data = file.evaluation;
+
+        /**
+         * NOTE: The filterControls array is used to specify which
+         * controls are selected.
+         * If we use the approach of filtering the content from the data object
+         * (e.g.
+         *   data.data.profiles[0].controls =
+         *     data.data.profiles[0].controls.filter((control) => {
+         *       if (filteredControls.includes(control.id)) {
+         *         return filteredControls.includes(control.id);
+         *       }
+         *     });
+         * )
+         * the contextualize object does not get updated and the results
+         * in data_store get out of sync, there is when utilizing the
+         * ".contains" it returns ann results where the file.evaluations
+         * returns the filtered controls.
+         */
+        let filteredControls: string[] = [];
```

Comment:
the way that this is implemented will not satisfy the issue where it was stated that we should be able to handle any of the filters that the user specifies.  this will only filter on pass/fail/whatever.

https://github.com/mitre/heimdall2/blob/dc5cddd7ae0f0c396044e91350044ff51ab00db0/apps/frontend/src/components/global/ExportCaat.vue#L43

use this example for how to get all the filter-passing controls.  it should also mean that we don't need basically all of what're currently lines 122-123 and 150-164 since it'll just be a simple assignment directly onto 'filteredControls' (which can also then be made const).

Reply:
Absolutely unequivocally disagree that the implementations does not satisfy the requirements. It works on all possible filtered permutations, did you even tried, apparently not if you're making this statement. 
_"the way that this is implemented will not satisfy the issue where it was stated that we should be able to handle any of the filters that the user specifies. this will only filter on pass/fail/whatever."_

Regarding the second opinion, the code that is being referenced (from the ExportCaat.vue) will not work. If you take a look at the reverse-html-mapper.ts it uses the file.data, which is the file.evaluation being set in ExportHTMLModule.vue and you are comparing to this code in the ExportCaat,vue `const data = file?.evaluation ?? '';`  now how does that filters the proper evaluations? it doesn't. The code on both are accomplishing different tasks.

Reply:
![image](https://github.com/mitre/heimdall2/assets/2015821/13ff94eb-bfd0-49e3-ac40-71ba06817d4e)

![image](https://github.com/mitre/heimdall2/assets/2015821/47d318d1-e640-427e-988a-8bd61c213920)

Here I've filtered it down to a single control using the nist filters.  The html export does not only include that single control.  Consequently, it does not meet the requirement of the original ticket according to my reading of it.  Like I said elsewhere, we should get @em-c-rod or @ejaronne's opinions on which of our interpretations is correct.

Reply:
w/r to your second paragraph, I'm only talking about extracting out the set of controls that pass the filtering.  You can use the store to get that set I'm hoping.

Reply:
This may be a use case, but the ask for this is just to be able to show only the failed controls (aka filter on status). If this becomes an additional ask, we can make another PR.

Reply:
I do think it is relevant to consider severity filters here. Looking at severity filters as well was not explicitly the ask, but seems like it is paired in the same use case.